src/ngx_http_lua_bodyfilterby.c

```diff
@@ -300,8 +344,21 @@ ngx_http_lua_body_filter(ngx_http_request_t *r, ngx_chain_t *in)
     lmcf = ngx_http_get_module_main_conf(r, ngx_http_lua_module);
     out = lmcf->body_filter_chain;
 
-    if (in == out) {
-        return ngx_http_next_body_filter(r, in);
+    if (in != out) {
+        /* content of body was replaced in ngx_http_lua_body_filter_param_set
+         * and the buffer was marked as consumed.
+         */
+        for (cl = in; cl != NULL; cl = ln) {
+            ln = cl->next;
+            if (ngx_buf_size(cl->buf) == 0) {
```

Comment:
This does not look right to me. 0-sized can be special bufs indicating flush, eof, and etc.